### PR TITLE
Update domains-faq.yml

### PR DESCRIPTION
**Pull request by msbemba**
Added Cloudflare to the list as it also supports All DNS records.

### CODE REVIEW

**Comment by msbemba** [comment]
issue#https://github.com/MicrosoftDocs/microsoft-365-docs/issues/6664

**Comment by msbemba** [comment]
@JohanFreelancer9 Please copy review

File: microsoft-365/admin/setup/domains-faq.yml [hunk 1]
```diff
@@ -81,6 +81,8 @@ sections:
           - eNomCentral
               
           - Europe Registry
+          
+          - CloudFlare
```
Comment (JohanFreelancer9) [inline microsoft-365/admin/setup/domains-faq.yml]:
```suggestion
          - Cloudflare
```

**Review by JohanFreelancer9 — CHANGES_REQUESTED** [review]
Copy review.

**Review by JohanFreelancer9 — APPROVED** [review]

**Comment by yogkumgit** [comment]
@scotvidican Please merge this PR. Thanks

**Comment by yogkumgit** [comment]
@pebaum Please merge this PR. Thanks

**Comment by yogkumgit** [comment]
@pebaum Gentle reminder.

**Pull request merged**